### ISSUE TITLE
Sobel Kernels in JND

### ISSUE THREAD

**Issue by BrianPulfer** [facebookresearch/active_indexing#3]
In file [activeindex/attenuations.py](https://github.com/facebookresearch/active_indexing/blob/main/activeindex/attenuations.py), in the `JND` class, there seems to be a mistake with the sobel kernels. In particular, the kernel for computing x and y gradients are the same:

```python
class JND(nn.Module):
    """ https://ieeexplore.ieee.org/document/7885108 """

    def __init__(self, preprocess = lambda x: x):
        super(JND, self).__init__()
        kernel_x = [[-1., 0., 1.], [-2., 0., 2.], [-1., 0., 1.]]
        kernel_y = [[-1., 0., 1.], [-2., 0., 2.], [-1., 0., 1.]]
```

They seem to both be kernels for the x-gradient. It also seems that at no point in time is the `kernel_y` transposed or used differently from `kernel_x`. Is this a bug or is this expected?

**Edit**
A quick fix would simply be: `kernel_y = [[1., 2., 1.], [0., 0., 0.], [-1., -2., -1.]]`

**Comment by pierrefdz** [comment]
Indeed, it seems unexpected... I will look into it and do the change you mentioned.

**Comment by pierrefdz** [comment]
Thanks @BrianPulfer for spotting the mistake. I made the changes and re-ran some experiments. 
This does not change the retrieval metrics, but changes the perceptual heatmap where gradients in the y-axis are strong (as expected).
